Sources/protoc-gen-swift/main.swift

```diff
@@ -27,8 +27,10 @@ import PluginLibrary
 enum GenerationError: Error {
   /// Raised for any errors reading the input
   case readFailure
-  /// Raise when parsing the parameter string and found an unknown key
+  /// Raised when parsing the parameter string and found an unknown key
   case unknownParameter(name: String)
+  /// Raised when a parameter was giving an invalid value
+  case invalidParameterValue(name: String, value: String)
```

Comment:
Apparently there's not automatic way, so Internet suggests adding this into the enum: `static let allValues = [value1, value2, ...]`, to be able and do `MyEnum.allValues`. It wouldn't avoid the repetition, although it'd keep it together with the original list.